sdk/core/System.ClientModel/samples/Diagnostics.md

```diff
@@ -1,6 +1,81 @@
+
+The following sample shows an example of instrumenting a service client implementation.
+
+```C# Snippet:OpenTelemetryInClient
+public class SampleClient
+{
+    private readonly Uri _endpoint;
+    private readonly ApiKeyCredential _credential;
+    private readonly ClientPipeline _pipeline;
+    private readonly SampleClientOptions _sampleClientOptions;
+
+    // Each client should have a static ActivitySource named after the full name
+    // of the client.
+    // Most of the time, tracing should start as experimental.
+    internal static ActivitySource ActivitySource { get; } = new($"Experimental.{typeof(MapsClient).FullName!}");
+
+    public SampleClient(Uri endpoint, ApiKeyCredential credential, SampleClientOptions? options = default)
+    {
+        options ??= new SampleClientOptions();
+        _sampleClientOptions = options;
+
+        _endpoint = endpoint;
+        _credential = credential;
+        ApiKeyAuthenticationPolicy authenticationPolicy = ApiKeyAuthenticationPolicy.CreateBearerAuthorizationPolicy(credential);
+
+        _pipeline = ClientPipeline.Create(options,
+            perCallPolicies: ReadOnlySpan<PipelinePolicy>.Empty,
+            perTryPolicies: new PipelinePolicy[] { authenticationPolicy },
+            beforeTransportPolicies: ReadOnlySpan<PipelinePolicy>.Empty);
+    }
+
+    public ClientResult<SampleResource> UpdateResource(SampleResource resource)
+    {
+        // Attempt to create and start an Activity for this operation.
+        // StartClientActivity does nothing and returns null if distributed tracing was
+        // disabled by the consuming application or if there are not any active listeners.
+        using Activity? activity = ActivitySource.StartClientActivity(_sampleClientOptions, $"{nameof(SampleClient)}.{nameof(UpdateResource)}");
+
+        try
+        {
+            using PipelineMessage message = _pipeline.CreateMessage();
+            PipelineRequest request = message.Request;
+            request.Method = "PATCH";
+            request.Uri = new Uri($"https://www.example.com/update?id={resource.Id}");
+            request.Headers.Add("Accept", "application/json");
+            request.Content = BinaryContent.Create(resource);
+
+            _pipeline.Send(message);
+
+            PipelineResponse response = message.Response!;
+            if (response.IsError)
+            {
+                throw new ClientResultException(response);
+            }
+            SampleResource updated = ModelReaderWriter.Read<SampleResource>(response.Content)!;
+            return ClientResult.FromValue(updated, response);
+        }
+        catch (Exception ex)
+        {
+            // Catch any exceptions and update the activity. Then re-throw the exception.
+            activity?.MarkFailed(ex);
+            throw;
+        }
+    }
+}
+```
+
```

Comment:
nit: can we add a basic example of how we expect end-users to enable distributed tracing for our libs?